tap-agent/src/agent/sender_account.rs

```diff
@@ -637,46 +647,50 @@ impl Actor for SenderAccount {
                 if should_deny {
                     state.add_to_denylist().await;
                 }
-                let total_counter_for_allocation = state
-                    .sender_fee_tracker
-                    .get_count_outside_buffer_for_allocation(&allocation_id);
-                let can_trigger_rav = state.sender_fee_tracker.can_trigger_rav(allocation_id);
-                let counter_greater_receipt_limit = total_counter_for_allocation
-                    >= state.config.tap.rav_request_receipt_limit
-                    && can_trigger_rav;
-                let total_fee_outside_buffer = state.sender_fee_tracker.get_ravable_total_fee();
-                let total_fee_greater_trigger_value =
-                    total_fee_outside_buffer >= state.config.tap.rav_request_trigger_value;
-                let rav_result = match (
-                    counter_greater_receipt_limit,
-                    total_fee_greater_trigger_value,
-                ) {
-                    (true, _) => {
-                        tracing::debug!(
-                            total_counter_for_allocation,
-                            rav_request_receipt_limit = state.config.tap.rav_request_receipt_limit,
-                            %allocation_id,
-                            "Total counter greater than the receipt limit per rav. Triggering RAV request"
-                        );
 
-                        state.rav_request_for_allocation(allocation_id).await
-                    }
-                    (_, true) => {
-                        tracing::debug!(
-                            total_fee_outside_buffer,
-                            trigger_value = state.config.tap.rav_request_trigger_value,
-                            "Total fee greater than the trigger value. Triggering RAV request"
+                let has_limit_available = state.adaptive_limiter.has_limit();
```

Comment:
I do not like much the name `has_limit_available`, id rather have something like `has_available_slots_for_requests` or thread or sth like that to make it more clear ?
What do you think 
I know its not parallel but maybe it gives the idea better